app/helpers/schema_helper.py

```diff
@@ -1,215 +1,177 @@
-from app.questionnaire.location import Location
 from app.validation.error_messages import error_messages
 
 
 class SchemaHelper(object):  # pylint: disable=too-many-public-methods
 
-    @staticmethod
-    def get_messages(survey_json):
-        messages = error_messages.copy()
+    def __init__(self, survey_json):
+        self.json = survey_json
 
-        if 'messages' in survey_json:
-            for key, message in survey_json['messages'].items():
-                messages[key] = message
-        return messages
+        self.messages = None
+        self.aliases = None
 
-    @staticmethod
-    def get_first_group_id(survey_json):
-        return survey_json['groups'][0]['id']
+        # Generate list of blocks
+        self.block_ids = []
+        for block in self.get_blocks():
+            self.block_ids.append(block.get('id'))
 
-    @classmethod
-    def get_first_block_id_for_group(cls, survey_json, group_id):
-        group = cls.get_group(survey_json, group_id)
-        if group:
-            return group['blocks'][0]['id']
+        self.blocks = {}
+        for block in self.get_blocks():
+            self.blocks[block.get('id')] = block
 
-    @classmethod
-    def is_first_block_id_for_group(cls, survey_json, group_id, block_id):
-        group = cls.get_group(survey_json, group_id)
-        return group is not None and group['blocks'][0]['id'] == block_id
+        # Generate list of Groups
+        self.group_ids = []
+        for group in self.get_groups():
+            self.group_ids.append(group.get('id'))
 
-    @staticmethod
-    def is_summary_or_confirmation(block):
-        return block and 'type' in block and block['type'] in ('Summary', 'Confirmation')
+        self.groups = {}
+        for group in self.get_groups():
+            self.groups[group.get('id')] = group
 
-    @staticmethod
-    def get_last_block_id(survey_json):
-        return survey_json['groups'][-1]['blocks'][-1]['id']
+    def get_messages(self):
+        if self.messages:
+            return self.messages
 
-    @staticmethod
-    def get_last_group_id(survey_json):
-        return survey_json['groups'][-1]['id']
+        self.messages = error_messages.copy()
 
-    @staticmethod
-    def get_last_block_in_group(group):
-        if group and 'blocks' in group:
-            return group['blocks'][-1]
+        if 'messages' in self.json:
+            for key, message in self.json['messages'].items():
+                self.messages[key] = message
```

Comment:
Can we do this in initialisation of the class, so we can remove this `get_messages` method and just use the messages property to get messages?